server/src/main/java/org/candlepin/controller/ManifestManager.java

```diff
@@ -0,0 +1,387 @@
+import org.candlepin.sync.file.ManifestFileType;
+import org.candlepin.sync.file.ManifestFileServiceException;
+import org.candlepin.util.Util;
+import org.quartz.JobDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xnap.commons.i18n.I18n;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+/**
+ * This class serves as a controller layer for manifest export and import and encapsulates
+ * the {@link Importer} and {@link Exporter} functionality so that it isn't spread out
+ * across multiple classes.
+ *
+ */
+public class ManifestManager {
+
+    private static Logger log = LoggerFactory.getLogger(ManifestManager.class);
+    private ManifestFileService manifestFileService;
+    private Exporter exporter;
+    private Importer importer;
+    private EntitlementCurator entitlementCurator;
+    private PoolManager poolManager;
+    private ConsumerCurator consumerCurator;
+    private PrincipalProvider principalProvider;
+    private I18n i18n;
+    private EventSink sink;
+    private EventFactory eventFactory;
+
+    @Inject
+    public ManifestManager(ManifestFileService manifestFileService, Exporter exporter, Importer importer,
+        ConsumerCurator consumerCurator, EntitlementCurator entitlementCurator, PoolManager poolManager,
+        PrincipalProvider principalProvider, I18n i18n, EventSink eventSink,
+        EventFactory eventFactory) {
+        this.manifestFileService = manifestFileService;
+        this.exporter = exporter;
+        this.importer = importer;
+        this.consumerCurator = consumerCurator;
+        this.entitlementCurator = entitlementCurator;
+        this.poolManager = poolManager;
+        this.principalProvider = principalProvider;
+        this.i18n = i18n;
+        this.sink = eventSink;
+        this.eventFactory = eventFactory;
+    }
+
+    /**
+     * Asynchronously generates a manifest for the target consumer.
+     *
+     * @param consumer the target consumer.
+     * @param cdnLabel
+     * @param webAppPrefix
+     * @param apiUrl
+     * @return the details of the async export job.
+     */
+    public JobDetail generateManifestAsync(Consumer consumer, String cdnLabel, String webAppPrefix,
+        String apiUrl) {
+        log.info("Scheduling Async Export for consumer {}", consumer.getUuid());
+        return ExportJob.scheduleExport(consumer, cdnLabel, webAppPrefix, apiUrl);
+    }
+
+    /**
+     * Generates a manifest for the specified consumer.
+     *
+     * @param consumer the target consumer.
+     * @param cdnLabel
+     * @param webAppPrefix
+     * @param apiUrl
+     * @return an archive of the target consumer
+     * @throws ExportCreationException when an export fails.
+     */
+    public File generateManifest(Consumer consumer, String cdnLabel, String webAppPrefix, String apiUrl)
+        throws ExportCreationException {
+        poolManager.regenerateDirtyEntitlements(entitlementCurator.listByConsumer(consumer));
```

Comment:
opportunity for optimization? perhaps we could fetch only dirty entitlements instead of fetching all of them and then checking if they are dirty?